### PR TITLE
feat: added loading text

### LINKED ISSUE #1725: Add loading animation for when playground generate models
#### Reason/Context

https://modelina.org/playground

When you change the input, open the playground, or change the settings, the playground need to ask the backend to re-generate the models, and there can be a significant delay here depending on input and latency to the backend server. Therefore I suggest we add a loading screen to the output models view that showcases that it is currently in the process of being generated.

You can find the most relevant code for the playground here: https://github.com/asyncapi/modelina/blob/master/modelina-website/src/components/playground/Playground.tsx

And the output panel can be found here: https://github.com/asyncapi/modelina/blob/master/modelina-website/src/components/playground/OutputNavigation.tsx

### PR DESCRIPTION
**Pull request by Shriya-Chauhan**
## Description
Added loader.

[Screencast from 11-30-2024 08:11:34 PM.webm](https://github.com/user-attachments/assets/dd5279ce-0af7-4eaa-ac41-0f5117cf5a08)

## Checklist
- [X] The code follows the project's coding standards and is properly linted (`npm run lint`).
- [X] Tests have been added or updated to cover the changes.
- [X] Documentation has been updated to reflect the changes.
- [ ] All tests pass successfully locally.(`npm run test`).

### CODE REVIEW

**Comment by Shriya-Chauhan** [comment]
@devilkiller-ag I have created pr for resolving issue #1725 . Please take a look and let me know if any changes required.

**Review by devilkiller-ag — CHANGES_REQUESTED** [review]

File: modelina-website/src/components/playground/Playground.tsx [hunk 1]
```diff
@@ -102,19 +106,22 @@ const Playground: React.FC<ModelinaPlaygroundProps> = (props) => {
...
             setStatusCode(200);
             setErrorMessage('');
+            setOutputLoading(false);
           })
           .catch((error) => {
             console.error(error);
             setError(true);
             setErrorMessage('Input is not a correct AsyncAPI document, so it cannot be processed.');
             setStatusCode(500);
+            setOutputLoading(true);
```
[the remark this answers is not in the document]
Comment (Shriya-Chauhan) [reply modelina-website/src/components/playground/Playground.tsx:116]: Ok! Working on this.

File: modelina-website/src/components/playground/Playground.tsx [hunk 2]
```diff
@@ -102,19 +106,22 @@ const Playground: React.FC<ModelinaPlaygroundProps> = (props) => {
...
           });
       }
     } catch (e: any) {
       console.error(e);
       setError(true);
       setErrorMessage('Input is not a correct AsyncAPI document, so it cannot be processed.');
       setStatusCode(400);
+      setOutputLoading(true);
```
[the remark this answers is not in the document]
Comment (Shriya-Chauhan) [reply modelina-website/src/components/playground/Playground.tsx]: Can you please review again. I have made the necessary changes @devilkiller-ag

**Comment by Shriya-Chauhan** [comment]
@devilkiller-ag Thank you for your support and guidance. I would love to contribute more—kindly point me to any other issues where I can help.

**Comment by devilkiller-ag** [comment]
> @devilkiller-ag Thank you for your support and guidance. I would love to contribute more—kindly point me to any other issues where I can help.

You can try #1360. Besides this, we have also some issues related to Modelina tool which you can explore on the [issues](https://github.com/asyncapi/modelina/issues/) page.

**Pull request merged**